internal/printer/printer.go

```diff
@@ -36,6 +36,9 @@ type printer struct {
 
 	// Optional, used only for TSX output
 	ranges TSXRanges
+	// Keep track of how many multi-byte characters we've printed so that they can be skipped whenever we need a character-based index
+	// This could be directly in the token / node information, however this would require a fairly large refactor
+	bytesToSkip int
```

Comment:
So, this not really optimal, because it confuse the printed bytes with the source bytes. It's fine in 99.999% of cases, because we're not printing any multi-bytes ourselves that were not part of the source. But it's something we *could* do and would have to take into account then.

The real fix, but it'd require a large amount of work would be to have this information directly on the tokens / nodes. I first tried to do that, but I quickly realised that I was changing *everything* and ran into the risk of breaking things everywhere. 

If this prove to be an issue, I'd definitely be willing to make the proper change, but in my testing this honestly proved to work just fine.